pkg/cassandra/config.go

```diff
@@ -75,32 +75,36 @@ func (c config) MarshalJSON() ([]byte, error) {
 	return json.Marshal(&config)
 }
 
-func newConfig(apiConfig *api.CassandraConfig, cassandraVersion string) config {
-	config := config{cassandraVersion: cassandraVersion}
+func newConfig(apiConfig *api.CassandraConfig, cassandraVersion string) (config, error) {
+	cfg := config{cassandraVersion: cassandraVersion}
+
+	if apiConfig == nil {
+		return config{}, DCConfigIncomplete{"CassandraConfig"}
+	}
 
 	if apiConfig.CassandraYaml == nil {
-		config.CassandraYaml = &api.CassandraYaml{}
+		cfg.CassandraYaml = &api.CassandraYaml{}
```

Comment:
Thanks for changing the variable names. It was definitely a bit confusing as was originally written. I'll pass that along to the author 😉

Reply:
@burmanm merging is blocked for me because of your requested changes. Can you please approve or talk to @jsanda  about whatever changes are required to get consensus with him.

I have manually tested and this is ready to go once Github gets rid of the 'changes requested' block.